### DISCUSSION TITLE
Pie-Charts for upstream DNS Servers and Query Types

### DISCUSSION THREAD

**Discussion by ctr-privat** [AdguardTeam/AdGuardHome#4883] [Ideas]
Hi, I'm switching from Pi-hole to AdGuard Home. 
I just wonder about the dns upstream server usage is not break down to any percentage or pie-chart overview. I choose the load-balancing mode for upstream DNS which should use some kind of "weighted random algorithm" but in the Query Log I can see that nearly just one of the 15 servers is used. I would like to have some kind of "overview/controll" about which upstream servers are used most or maby if only one is used.
Also I would appreciate a pie-chart for the query types like "A (IPv4)", "AAAA (IPv6)", "PTR", etc. Just like my old pihole does... (it has ring-charts for these)

Would be great to get some feedback from you, thanks in advance :)

**Comment by ainar-g** [comment]
Hello.  Better statistics is one of the things planned for the v0.108 dev cycle.  See #2554 and the related issues.

I'm not sure about pie charts, since they are generally considered bad design, but we'll see.

**Reply by sinkingfeeling** [reply]
@ainar-g In the meantime is there any for us to see upstream DNS usage / spread?

**Reply by ainar-g** [reply]
Currently, this is possible if you analyse the querylog file.  For example by taking the `querylog.json` file and summing up records based on the `Upstream` field.
